tests/infra_test.go

```diff
@@ -95,6 +99,83 @@ var _ = Describe("[Serial][sig-compute]Infrastructure", func() {
+			Expect(err).ToNot(HaveOccurred())
+			start := time.Now()
+			libreplicaset.DoScaleWithScaleSubresource(virtClient, replicaset.Name, 10)
+			fastDuration := time.Now().Sub(start)
+			libreplicaset.DoScaleWithScaleSubresource(virtClient, replicaset.Name, 0)
+
+			By("reducing the throughput on controller")
+			originalKubeVirt := util.GetCurrentKv(virtClient)
+			originalKubeVirt.Spec.Configuration.ControllerConfiguration = &v1.ReloadableComponentConfiguration{
+				RestClient: &v1.RESTClientConfiguration{
+					RateLimiter: &v1.RateLimiter{
+						TokenBucketRateLimiter: &v1.TokenBucketRateLimiter{
+							Burst: 3,
+							QPS:   2,
+						},
+					},
+				},
+			}
+			tests.UpdateKubeVirtConfigValueAndWait(originalKubeVirt.Spec.Configuration)
+			By("starting a replicaset with reduced throughput")
+			start = time.Now()
+			libreplicaset.DoScaleWithScaleSubresource(virtClient, replicaset.Name, 10)
+			slowDuration := time.Now().Sub(start)
+			Expect(slowDuration.Seconds()).To(BeNumerically(">", 5*fastDuration.Seconds()))
+		})
+
+		It("on the virt handler rate limiter should lead to delayed VMI running states", func() {
+			By("first getting the basetime for a replicaset")
+			targetNode := util.GetAllSchedulableNodes(virtClient).Items[0]
+			vmi := libvmi.NewCirros(
+				libvmi.WithResourceMemory("1Mi"),
+				libvmi.WithNodeSelectorFor(&targetNode),
+			)
+			replicaset := tests.NewRandomReplicaSetFromVMI(vmi, 0)
+			replicaset, err = virtClient.ReplicaSet(util.NamespaceTestDefault).Create(replicaset)
+			Expect(err).ToNot(HaveOccurred())
+			start := time.Now()
+			libreplicaset.DoScaleWithScaleSubresource(virtClient, replicaset.Name, 10)
+			Eventually(matcher.AllVMIs(replicaset.Namespace), 90*time.Second, 1*time.Second).Should(matcher.BeInPhase(v1.Running))
+			fastDuration := time.Now().Sub(start)
+
+			libreplicaset.DoScaleWithScaleSubresource(virtClient, replicaset.Name, 0)
+			Eventually(matcher.AllVMIs(replicaset.Namespace), 90*time.Second, 1*time.Second).Should(matcher.BeGone())
+
+			By("reducing the throughput on handler")
+			originalKubeVirt := util.GetCurrentKv(virtClient)
+			originalKubeVirt.Spec.Configuration.HandlerConfiguration = &v1.ReloadableComponentConfiguration{
+				RestClient: &v1.RESTClientConfiguration{
+					RateLimiter: &v1.RateLimiter{
+						TokenBucketRateLimiter: &v1.TokenBucketRateLimiter{
+							Burst: 1,
+							QPS:   1,
+						},
+					},
+				},
+			}
+			tests.UpdateKubeVirtConfigValueAndWait(originalKubeVirt.Spec.Configuration)
+
+			By("starting a replicaset with reduced throughput")
+			start = time.Now()
+			libreplicaset.DoScaleWithScaleSubresource(virtClient, replicaset.Name, 10)
+			Eventually(matcher.AllVMIs(replicaset.Namespace), 180*time.Second, 1*time.Second).Should(matcher.BeInPhase(v1.Running))
+			slowDuration := time.Now().Sub(start)
+			Expect(slowDuration.Seconds()).To(BeNumerically(">", 1.5*fastDuration.Seconds()))
```

Comment:
Is there any reason why you expect the rate limit make the latency 50% slower?
Instead of relative times, it might be better to have absolute times, as used in the first test.

Reply:
Maybe this is where the flaky test that David pointed before is.
The first test `Expect(slowDuration.Seconds()).To(BeNumerically(">", 5*fastDuration.Seconds()))` you used the relative threshold as 5 times higher than the fastDuration; the second test is 1.5 times.
If you prefer to use relative threshold instead of absolute ones, maybe both tests should have similar thresholds?

Reply:
I have to make it relative. Otherwise the tests could never succeed locally and in CI. I choose here only 1.5 since the difference from create-to-running is from a factor perspective less big when throttling only scheduled-to-running, vs created-to-scheduled above.

Reply:
I saw that you have updated how the `fastDuration` is measured.
Looks good!